src/dmd/expression.d

```diff
@@ -1531,6 +1537,47 @@ StringExp semanticString(Scope *sc, Expression exp, const char* s)
     return se;
 }
 
+/***************************************************
+ * Given an Expression, find the variable it really is.
+ * for example, `a[index]` is really `a`, and `s.f` is really `s`.
+ * Params:
+ *      e = Expression to look at
+ * Returns:
+ *      variable if there is one, null if not
+ */
+VarDeclaration expToVariable(Expression e)
+{
+    while (1)
```

Comment:
`while(true)`

Reply:
Yeah, I tried that once, too:  https://github.com/dlang/dmd/pull/7861#discussion_r167387243